src/runtime_src/core/edge/ps_kernels/built_in/instance_query/instance_query.c

```diff
@@ -0,0 +1,272 @@
+#include "xrt/xrt_kernel.h"
+
+#include <chrono>
+#include <cstring>
+#include <ctime>
+#include <fstream>
+#include <map>
+#include <set>
+#include <syslog.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/utsname.h>
+#include <vector>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/property_tree/ini_parser.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+namespace pt = boost::property_tree;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
I think only the entry points need to be extern "C".    Maybe re-organize and be more explicit about which functions are exported as C functions and which are internal and C++.

Reply:
I moved the extern to only included the function you mentioned below